### PR TITLE
add shell scripts in mno analysis tools dir

### PR DESCRIPTION
**Pull request by DanielMwendwa**
review #92 first

### CODE REVIEW

File: mno_analysis_tools/docker-run-compute-msg-difference-btwn-two-firebase-time-periods.sh [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+fi
+
+if [[ -z $7 ]]; then
+OUTPUT_DIR=$6
+CMD="pipenv run python -u compute_msg_difference_btwn_two_firebase_time_periods.py \
+    /data/raw_messages.json /data/${MSG_DIRECTION}_msg_diff_per_period.json \
+    \"$TARGET_OPERATOR\" \"$TARGET_MESSAGE_DIRECTION\"  \"$START_DATE\" \"$END_DATE\" 
+"
+else
+TIME_FRAME=$6
```
Comment (IsaackMwenda) [inline mno_analysis_tools/docker-run-compute-msg-difference-btwn-two-firebase-time-periods.sh:39]: put all the arguments in one block

File: mno_analysis_tools/docker-run-compute-messages-per-period.sh [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+# Copy input data into the container
+docker cp "$RAW_MESSAGES_FILE_PATH" "$container:/data/raw_messages.json"
+
+# Run the container
+docker start -a -i "$container"
+
+# Copy the output data back out of the container
+echo "Copying $container_short_id:/data/. -> $OUTPUT_DIR"
+# mkdir -p "$OUTPUT_DIR"
```
Comment (IsaackMwenda) [inline mno_analysis_tools/docker-run-compute-messages-per-period.sh:57]:
```suggestion
mkdir -p "$OUTPUT_DIR"
```

File: mno_analysis_tools/docker-run-compute-msg-difference-btwn-two-firebase-time-periods.sh [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+# Copy input data into the container
+docker cp "$RAW_MESSAGES_FILE_PATH" "$container:/data/raw_messages.json"
+
+# Run the container
+docker start -a -i "$container"
+
+# Copy the output data back out of the container
+echo "Copying $container_short_id:/data/. -> $OUTPUT_DIR"
+# mkdir -p "$OUTPUT_DIR"
```
Comment (IsaackMwenda) [inline mno_analysis_tools/docker-run-compute-msg-difference-btwn-two-firebase-time-periods.sh:65]: ditto

File: mno_analysis_tools/docker-run-compute-window-of-downtime.sh [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+# Copy input data into the container
+docker cp "$RAW_MESSAGES_FILE_PATH" "$container:/data/raw_messages.json"
+
+# Run the container
+docker start -a -i "$container"
+
+# Copy the output data back out of the container
+echo "Copying $container_short_id:/data/. -> $OUTPUT_DIR"
+# mkdir -p "$OUTPUT_DIR"
```
Comment (IsaackMwenda) [inline mno_analysis_tools/docker-run-compute-window-of-downtime.sh:56]: ditto

File: mno_analysis_tools/docker-run-compute-messages-per-period.sh (hunk 2 above)
Comment (DanielMwendwa) [reply mno_analysis_tools/docker-run-compute-messages-per-period.sh:57]: Updated thanks.

File: mno_analysis_tools/docker-run-compute-window-of-downtime.sh (hunk 4 above)
Comment (DanielMwendwa) [reply mno_analysis_tools/docker-run-compute-window-of-downtime.sh]: Updated thanks.

File: mno_analysis_tools/docker-run-compute-msg-difference-btwn-two-firebase-time-periods.sh (hunk 3 above)
Comment (DanielMwendwa) [reply mno_analysis_tools/docker-run-compute-msg-difference-btwn-two-firebase-time-periods.sh]: Updated thanks.

File: mno_analysis_tools/docker-run-compute-msg-difference-btwn-two-firebase-time-periods.sh (hunk 1 above)
Comment (DanielMwendwa) [reply mno_analysis_tools/docker-run-compute-msg-difference-btwn-two-firebase-time-periods.sh:41]: Updated thanks.

File: mno_analysis_tools/docker-run-compute-messages-per-period.sh [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+# Copy the output data back out of the container
+echo "Copying $container_short_id:/data/. -> $OUTPUT_DIR"
+mkdir -p "$OUTPUT_DIR"
+docker cp "$container:/data/." "$OUTPUT_DIR"
+
+if [[ "$PROFILE_MEMORY" = true ]]; then
+    echo "Copying $container_short_id:/data/memory.prof -> $MEMORY_PROFILE_OUTPUT_PATH"
+    mkdir -p "$(dirname "$MEMORY_PROFILE_OUTPUT_PATH")"
+    docker cp "$container:/data/memory.prof" "$MEMORY_PROFILE_OUTPUT_PATH"
+fi
```
Comment (IsaackMwenda) [inline mno_analysis_tools/docker-run-compute-messages-per-period.sh:83]: # Tear down the container, now that all expected output files have been copied out successfully
docker container rm "$container" >/dev/null

File: mno_analysis_tools/docker-run-compute-messages-per-period.sh [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    # Tear down the container when done.
+    docker container rm "$container" >/dev/null
+}
+trap finish EXIT
+
+# Copy input data into the container
+docker cp "$RAW_MESSAGES_FILE_PATH" "$container:/data/raw_messages.json"
+
+# Run the container
+docker start -a -i "$container"
```
Comment (IsaackMwenda) [inline mno_analysis_tools/docker-run-compute-messages-per-period.sh:72]: log container id before starting for easy debugging
`echo "Starting container $container_short_id"`

File: mno_analysis_tools/docker-run-fetch-raw-messages.sh [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+# Copy the output data back out of the container
+echo "Copying $container_short_id:/data/. -> $OUTPUT_DIR"
+mkdir -p "$OUTPUT_DIR"
+docker cp "$container:/data/." "$OUTPUT_DIR"
+
+if [[ "$PROFILE_MEMORY" = true ]]; then
+    echo "Copying $container_short_id:/data/memory.prof -> $MEMORY_PROFILE_OUTPUT_PATH"
+    mkdir -p "$(dirname "$MEMORY_PROFILE_OUTPUT_PATH")"
+    docker cp "$container:/data/memory.prof" "$MEMORY_PROFILE_OUTPUT_PATH"
+fi
```
Comment (IsaackMwenda) [inline mno_analysis_tools/docker-run-fetch-raw-messages.sh:68]: tear down containers

File: mno_analysis_tools/docker-run-compute-messages-per-period.sh (hunk 6 above)
Comment (DanielMwendwa) [reply mno_analysis_tools/docker-run-compute-messages-per-period.sh:73]: Updated thanks

File: mno_analysis_tools/docker-run-compute-messages-per-period.sh (hunk 5 above)
Comment (DanielMwendwa) [reply mno_analysis_tools/docker-run-compute-messages-per-period.sh:84]: I've written this function that is invoked on exit
```
function finish {
    # Tear down the container when done.
    docker container rm "$container" >/dev/null
}
```

**Comment by DanielMwendwa** [comment]
> Thanks, we now only need one run script that runs the rest of the scripts

I'll open another PR

File: mno_analysis_tools/docker-run-compute-messages-per-period.sh (hunk 5 above)
Comment (IsaackMwenda) [reply mno_analysis_tools/docker-run-compute-messages-per-period.sh:84]: well there is no need for a function just call executing the statement is enough

**Pull request merged**